intake_esm/cmip.py

```diff
@@ -225,31 +213,71 @@ def results(self):
             self.query_results = get_subset(self.collection_name, self.collection_type, self.query)
             return self.query_results
 
+    def to_xarray(self, **kwargs):
+        """Return dataset as an xarray instance"""
+        _kwargs = self.kwargs.copy()
+        _kwargs.update(kwargs)
+        self.kwargs = _kwargs
+        return self.to_dask()
+
     def _open_dataset(self):
 
-        kwargs = self._kwargs
-        if 'concat_dim' not in kwargs.keys():
-            kwargs.update(concat_dim=self.concat_dim)
-        if self.pattern:
-            kwargs.update(preprocess=self._add_path_to_ds)
+        kwargs = self.kwargs
+        if self.query_results.empty:
+            raise ValueError(f'Query={self.query} returned empty results')
+
+        query = dict(self.query)
+
+        if 'decode_times' not in kwargs.keys():
+            kwargs.update(decode_times=False)
+        if 'time_coord_name' not in kwargs.keys():
+            kwargs.update(time_coord_name='time')
+        if 'ensemble_dim_name' not in kwargs.keys():
+            kwargs.update(ensemble_dim_name='member_id')
+        if 'chunks' not in kwargs.keys():
+            kwargs.update(chunks=None)
```

Comment:
This code is duplicate that in `cesm.py`. Are there ways we can avoid this duplication? As things evolve, identical implementations might be required in two places.